src/components/Courses/components/CourseCard/tests/courseCard.test.js

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { Provider } from "react-redux";
+import { configureStore } from "@reduxjs/toolkit";
+import { MemoryRouter } from "react-router-dom";
+import CourseCard from "../CourseCard";
+import rootReducer from "../../../../../store/rootReducer";
+import { ROLES } from "../../../../../constants";
+import formatDuration from "../../../../../helpers/formatDuration";
+import dateFormater from "../../../../../helpers/dateFormatter";
+
+// Helper function to render with Redux and Router
+const renderWithRedux = (
```

Comment:
You can extract this function to make your test files smaller and reuse it in all tests.
https://github.com/chacon44/React/pulls

Reply:
done